src/app/todos/page.js

```diff
@@ -36,7 +36,19 @@ export default function ToDos() {
     }
 
     function removeTodo({ index }) {
+        //get the id of the todo that was deleted
+        const id = todos[index].id;
+        console.log(id);
         setTodos(todos.filter((v,idx) => idx!==index));
+        fetch(`/api/todos/${id}`, { method: "delete" }).then((response) => response.ok && response.json());
+    }
+
+    function updateTodo(e) {
+        const id = todos[e].id;
+        todos[e].done = !todos[e].done;
+        console.log(id);
+        setTodos([...todos]);
```

Comment:
move this logic into the a then of the changed promise so it only happens on success